src/vm/gchelpers.cpp

```diff
@@ -554,11 +714,18 @@ OBJECTREF AllocateArrayEx(MethodTable *pArrayMT, INT32 *pArgs, DWORD dwNumArgs,
         ThrowOutOfMemoryDimensionsExceeded();
 
     // Allocate the space from the GC heap
-    S_SIZE_T safeTotalSize = S_SIZE_T(cElements) * S_SIZE_T(componentSize) + S_SIZE_T(pArrayMT->GetBaseSize());
+#ifdef _TARGET_64BIT_
```

Comment:
Is this ifdef worth it? This whole path is pretty slow. I doubt that saving a few instruction will make a difference.

Reply:
It mattered in the "Sz" helper. According to VTune the piece of safe math was costing more than anything else in the method.  
Without knowledge that we deal with 32bit values or smaller, safe helpers assumed worst case, so we had 3 uninlined calls taking structs packed in xmm registers and what not. There was not much of compile time short-circuiting.

If running on 64bit was a rare special case, I'd be ok with just leaving it as is, but since it is the 32bit that becomes rare nowdays (and only 32bit needs safe math here), I think `#ifdef` is worth it.

And then I copied the pattern to `Ex` helper, just to be consistent with `Sz`. It may make less difference here, just for consistency.

---
In reply to: [279988887](https://github.com/dotnet/coreclr/pull/24096#discussion_r279988887) [](ancestors = 279988887)

Reply:
I assume we keep the `ifdef` here since this is specialization for 64bit.

32bit specialization (the fastStringAllocator and the like) is gone though.

---
In reply to: [280143708](https://github.com/dotnet/coreclr/pull/24096#discussion_r280143708) [](ancestors = 280143708,279988887)

Reply:
ok